account_asset_management/models/account_asset.py

```diff
@@ -388,10 +390,12 @@ def _check_method(self):
                 _("Degressive-Linear is only supported for Time Method = Year.")
             )
 
-    @api.constrains("date_start", "method_end", "method_time")
+    @api.constrains("date_start", "method_end", "method_number", "method_time")
     def _check_dates(self):
         if self.filtered(
-            lambda a: a.method_time == "end" and a.method_end <= a.date_start
+            lambda a: a.method_time == "year"
+            and not a.method_number
+            and a.method_end <= a.date_start
```

Comment:
Looks like this might have caused the test failures
```
  File "/home/travis/build/OCA/account-financial-tools/account_asset_management/models/account_asset.py", line 398, in <lambda>
    and a.method_end <= a.date_start
TypeError: '<=' not supported between instances of 'bool' and 'datetime.date'
```

Reply:
solved